cobalt/renderer/cobalt_render_frame_observer.h

```diff
@@ -0,0 +1,39 @@
+// Copyright 2025 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef COBALT_RENDERER_COBALT_RENDERER_FRAME_OBSERVER_H_
+#define COBALT_RENDERER_COBALT_RENDERER_FRAME_OBSERVER_H_
+
+#include "content/public/renderer/render_frame.h"
+#include "content/public/renderer/render_frame_observer.h"
+
+namespace cobalt {
+
+class CobaltRenderFrameObserver : public content::RenderFrameObserver {
```

Comment:
Can we add a comment here? Mostly about what RenderFrameObserver does and why do we override it. E.g. https://source.corp.google.com/h/lbshell-internal/cobalt_src/+/main:cobalt/browser/cobalt_content_browser_client.h;l=34-39;drc=b4be47049c89d5140b1ac12db696ff71be5547dd .